spyderlib/spyder.py

```diff
@@ -2660,6 +2674,93 @@ def show_tour(self, index):
+        latest_release = self.worker_updates.latest_release
+        feedback = self.worker_updates.feedback
+        error_msg = self.worker_updates.error
+
+        url_r = 'https://github.com/spyder-ide/spyder/releases'
+        url_i = 'http://pythonhosted.org/spyder/installation.html'
+
+        # Define the custom QMessageBox
+        box = MessageCheckBox()
+        box.setWindowTitle(_("Spyder updates"))
+        box.set_checkbox_text(_("Check for updates on startup"))
+        box.setStandardButtons(QMessageBox.Ok)
+        box.setDefaultButton(QMessageBox.Ok)
+        box.setIcon(QMessageBox.Information)
+
+        # Adjust the checkbox depending on the stored configuration
+        section, option = 'main', 'check_updates_on_startup'
+        check_updates = CONF.get(section, option)
+        box.set_checked(check_updates)
+
+        if error_msg is not None:
+            msg = error_msg
+            box.setText(msg)
+            box.set_check_visible(False)
+            box.exec_()
+            check_updates = box.is_checked()
+        else:
+            if update_available:
+                msg = _("<b>Spyder {0} is available!</b> <br><br>Please use "
+                        "your package manager to update Spyder or go to our "
+                        "<a href=\"{1}\">Releases</a> page to download this "
+                        "new version. <br><br>If you are not sure how to "
+                        "proceed to update Spyder please refer to our "
+                        " <a href=\"{2}\">Installation</a> instructions."
+                        "" .format(latest_release, url_r, url_i))
+                box.setText(msg)
+                box.set_check_visible(True)
+                box.exec_()
+                check_updates = box.is_checked()
+            elif feedback:
+                msg = _("Spyder is up to date.")
+                box.setText(msg)
+                box.set_check_visible(False)
+                box.exec_()
+                check_updates = box.is_checked()
+
+        # Update checkbox based on user interaction
+        CONF.set(section, option, check_updates)
+
+        # Enable check_updates_action after the thread has finished
+        self.check_updates_action.setDisabled(False)
+
+    def check_updates(self):
+        """
+        Check for spyder updates on github releases using a QThread.
+        """
+        from spyderlib.workers.updates import WorkerUpdates
+
+        # Disable check_updates_action while the thread is working
+        self.check_updates_action.setDisabled(True)
+
+        # feedback` = False is used on startup, so only positive feedback is
+        # given. `feedback` = True is used when using the menu action, and
+        # gives feeback if updates are, or are not found.
+        feedback = False
+
+        if DEV:
+            feedback = True
+        elif self.thread_updates is not None:
+            self.thread_updates.terminate()
+            feedback = True

```

Comment:
I don't feel at ease with this `feedback` variable. It's passed to the worker, but it doesn't do anything with it. It seems the worker just holds it, so that `_check_updates_ready` can later read it.

But wouldn't it be better to move this block to `_check_updates_ready` directly and get rid of this passing around process?